misc/tests/test-parameters.py

```diff
@@ -0,0 +1,258 @@
+    s = str(subprocess.run(["./../../builds/release/bin/downward",
+                            "--help", "--txt2tags",
+                            "{}".format(feature_name)],
+                           stdout=subprocess.PIPE).stdout)
+    position = s.find(feature_name + "(")
+    assert position != -1
+    s = s[position + len(feature_name) + 1::] # start after the first '('
+    stack = ['(']
+    result = []
+    for c in s:
+        if c == '(' or c == "[":
+            stack.append(c)
+        elif c == ')' or c == "]":
+            assert matching(stack[-1], c)
+            stack.pop()
+            if not stack:
+                break
+        if len(stack) == 1: # not within nested parenthesis/brackets
+            result.append(c)
+    result = ''.join(result)
+    result = re.sub(r'=.*?,', ',', result + ",").split()
+    return result
+
+def extract_feature_name_and_cpp_class(cc_file, cc_files, cwd, num):
+    source_without_comments = subprocess.check_output(
+        ["gcc", "-fpreprocessed", "-dD", "-E", cc_file]).decode("utf-8")
+
+    name_pattern = r'TypedFeature\("([^"]*)"\)'
+    class_pattern = r'TypedFeature<(.*?)>'
+
+    feature_names = []
+    class_names = []
+    other_namespaces = []
+    feature_error_msgs = []
+    class_error_msgs = []
+    for line in source_without_comments.splitlines():
+        if re.search(name_pattern, line):
+            feature_name = re.search(name_pattern, line).group(1)
+            feature_error_msg = "feature_name: " + feature_name + "\n"
+            feature_names.append(feature_name)
+            feature_error_msgs.append(feature_error_msg)
+
+        if re.search(class_pattern, line):
+            feature_class = re.search(class_pattern, line).group(1)
+            class_name = feature_class.split()[-1].split("::")[-1]
+            other_namespace = (len(feature_class.split()[-1].split("::"))
+                               == 2)
+            class_error_msg = "class_name: " + class_name + "\n"
+            class_names.append(class_name)
+            other_namespaces.append(other_namespace)
+            class_error_msgs.append(class_error_msg)
+    return (feature_names[num], class_names[num], other_namespaces[num],
+           feature_error_msgs[num] + class_error_msgs[num])
+
+def get_cpp_class_parameters(
+        class_name, other_namespace, cc_file, cc_files, cwd):
+    found_in_file, parameters = get_constructor_parameters(
+        cc_file, class_name)
+    if not found_in_file:
+        # check in all files
+        for cc_file2 in cc_files:
+            found_in_file, parameters = get_constructor_parameters(
+                cc_file2, class_name)
+            if found_in_file:
+                break
+
+    if found_in_file:
+        parameters = parameters.replace("\n", "") + ","
+        parameters = parameters.split()
+        parameters = [word for word in parameters if "," in word]
+        parameters = [re.sub(C_VAR_PATTERN, '', word) + ","
```

Comment:
Yes I just meant appending the comma at the end. I think I was confused because the commas should be there already but I see now that the `C_VAR_PATTERN` strips them away again. So all good :).